### PR TITLE
Dpp 1283 duplicate maths and english payments

### PR DESCRIPTION
**Pull request by davehowlett**

### CODE REVIEW

File: src/PeriodEnd/PaymentsDue/DeployPaymentsDue/sql/dml/14 PeriodEnd.PaymentsDue.PreRun.Staging.ApprenticeshipEarnings3.sql [hunk 1]
```diff
@@ -74,21 +74,16 @@ INSERT INTO Staging.ApprenticeshipEarnings3
...
+			 ae.EpisodeStartDate >= (
+    Select
+    Case WHEN  [Name] = 'R01' OR [Name] = 'R02' OR [Name] = 'R03' OR [Name] = 'R04' OR [Name] = 'R05'  THEN CONVERT(VARCHAR(10), '08/01/' +  Cast(CalendarYear as varchar) , 101) 
+        ELSE CONVERT(VARCHAR(10), '08/01/' +  Cast(CalendarYear -1  as varchar) , 101) END
+        From  Reference.CollectionPeriods Where [Open] = 1)
+    AND
+        ae.EpisodeStartDate <= ( Select 
+        Case WHEN  [Name] = 'R01' OR [Name] = 'R02' OR [Name] = 'R03' OR [Name] = 'R04' OR [Name] = 'R05'  THEN CONVERT(VARCHAR(10), '07/31/' +  Cast(CalendarYear +1 as varchar) , 101) 
+        ELSE CONVERT(VARCHAR(10), '07/31/' +  Cast(CalendarYear as varchar) , 101) END
+        From  Reference.CollectionPeriods Where [Open] = 1)
```
Comment (scottywakefield) [inline src/PeriodEnd/PaymentsDue/DeployPaymentsDue/sql/dml/14 PeriodEnd.PaymentsDue.PreRun.Staging.ApprenticeshipEarnings3.sql:87]: Question: would this case statement be better as a function in sql?

**Review by scottywakefield — APPROVED** [review]

File: src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Features/Ilr Changes/learner_changes_contract_type.feature [hunk 2]
```diff
@@ -185,4 +185,75 @@ Scenario: DPP_965_05 - Non-Levy apprentice, provider edits contract type (ACT) i
...
+        | learner a | programme only DAS |              | 06/08/2017 | 20/08/2018       |                 | continuing        | maths or english | 1                   | 471      | 403            | 2              | 1            | DAS           | 06/08/2017              |
+        | learner a | programme only DAS | 9000         | 06/08/2017 | 20/08/2018       |                 | continuing        | programme        | 2                   |          | 403            | 2              | 1            |               |                         |
+		
+	Then the provider earnings and payments break down as follows:
+        | Type                                    | 08/17  | 09/17  | 10/17  | 11/17  | 12/17  | 01/18  | 02/18  | 03/18  | 04/18   | 05/18  |
+        | Provider Earned Total                   | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25  | 639.25 |
+        | Provider Earned from SFA                | 0      | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25  | 639.25 |
```
Comment (Pauljgraham) [inline src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Features/Ilr Changes/learner_changes_contract_type.feature]: Provider Earned from SFA should not be offset (only payments are offset)

File: src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Features/Ilr Changes/learner_changes_contract_type.feature [hunk 3]
```diff
@@ -185,4 +185,75 @@ Scenario: DPP_965_05 - Non-Levy apprentice, provider edits contract type (ACT) i
...
+        | learner a | programme only Non-DAS |              | 06/08/2017 | 20/08/2018       |                 | continuing        | maths or english | 1                   | 471      | 403            | 2              | 1            | Non-DAS       | 06/08/2017              |
+        | learner a | programme only Non-DAS | 9000         | 06/08/2017 | 20/08/2018       |                 | continuing        | programme        | 2                   |          | 403            | 2              | 1            |               |                         |
+		
+	Then the provider earnings and payments break down as follows:
+        | Type                                    | 08/17  | 09/17  | 10/17  | 11/17  | 12/17  | 01/18  | 02/18  | 03/18  | 04/18   | 05/18  |
+        | Provider Earned Total                   | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25  | 639.25 |
+        | Provider Earned from SFA                | 0      | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25 | 639.25  | 639.25 |
```
Comment (Pauljgraham) [inline src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Features/Ilr Changes/learner_changes_contract_type.feature]: Provider earned from SFA should not be offset. Althrough no assertion is done on this line

File: src/PeriodEnd/PaymentsDue/DeployPaymentsDue/sql/dml/14 PeriodEnd.PaymentsDue.PreRun.Staging.ApprenticeshipEarnings3.sql (hunk 1 above)
Comment (Pauljgraham) [reply src/PeriodEnd/PaymentsDue/DeployPaymentsDue/sql/dml/14 PeriodEnd.PaymentsDue.PreRun.Staging.ApprenticeshipEarnings3.sql:87]: Could we add a comment to this so it says it's checking episode start date is in the academic year.

**Review by Pauljgraham — CHANGES_REQUESTED** [review]

**Review by Pauljgraham — APPROVED** [review]

**Review by carlbm — APPROVED** [review]

**Comment by satya-sanikommu** [comment]
Review and testing completed. Ready to merge

**Pull request merged**